jetty-client/src/main/java/org/eclipse/jetty/client/AbstractConnectionPool.java

```diff
@@ -223,8 +223,8 @@ public void succeeded(Connection connection)
             {
                 if (LOG.isDebugEnabled())
                     LOG.debug("Connection {}/{} creation succeeded {}", pool.size(), pool.getMaxEntries(), connection);
-                adopt(entry, connection);
-                future.complete(null);
+                adopt(reservation, connection);
+                future.complete(null);  // TODO could this now be a future that passes back the connection?
```

Comment:
Remove the TODO, as `ConnectionPool` does not work in this way: by the time we notify the future, the `Connection` could already be stolen by another thread.